unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -0,0 +1,310 @@
+using Cysharp.Threading.Tasks;
+using DCL;
+using DCl.Social.Friends;
+using DCL.Social.Friends;
+using Decentraland.Renderer.RendererServices;
+using Decentraland.Social.Friendships;
+using Google.Protobuf.WellKnownTypes;
+using JetBrains.Annotations;
+using RPC;
+using System;
+using System.Threading;
+using rpc_csharp;
+using rpc_csharp.transport;
+using RPC.Transports;
+using System.Collections.Generic;
+using UnityEngine;
+using Payload = Decentraland.Social.Friendships.Payload;
+
+namespace MainScripts.DCL.Controllers.FriendsController
+{
+    public class RPCSocialApiBridge : IRPCSocialApiBridge
+    {
+        private const int REQUEST_TIMEOUT = 30;
+
+        private string accessToken;
+
+        private readonly IUserProfileBridge userProfileWebInterfaceBridge;
+
+        public event Action<UserStatus> OnFriendAdded;
+        public event Action<string> OnFriendRemoved;
+        public event Action<FriendRequest> OnFriendRequestAdded;
+        public event Action<string> OnFriendRequestRemoved;
+        public event Action<AddFriendRequestsPayload> OnFriendRequestsAdded;
+        public event Action<FriendshipUpdateStatusMessage> OnFriendshipStatusUpdated;
+        public event Action<FriendRequestPayload> OnFriendRequestReceived;
+
+        private ClientFriendshipsService socialClient;
+
+        public RPCSocialApiBridge(MatrixInitializationBridge matrixInitializationBridge, IUserProfileBridge userProfileWebInterfaceBridge)
+        {
+            matrixInitializationBridge.OnReceiveMatrixAccessToken += (token) => { this.accessToken = token; };
+            this.userProfileWebInterfaceBridge = userProfileWebInterfaceBridge;
+        }
+
+        public async UniTaskVoid InitializeClient(CancellationToken cancellationToken = default)
```

Comment:
It is not clear why we are forgetting about the UniTask as other code relies on the fields set

Reply:
I don't know exactly what you're referring to, if you're referring to why the subscriptions are being forgotten, then it's because in the future, the subscriptions will be awaited for processes that will never end until the application shuts down.
If you're referring to the controller calling `socialApiBridge.InitializeClient(controllerCancellationTokenSource.Token).Forget();` then it's because the `Initialize` is not an async method.

Reply:
Considering the initialization flow of `RPCSocialApiBridge` I highly recommend abstracting from what is currently written and trying to revise it:

- There are obvious initialization dependencies that should fire once before any of the methods can be ever called
- It's obscure and inefficient to wait for them by repetitive `WaitUntil`
- Waiting for the initialization should be done via a single point of synchronization, for example how it is done for `AddressablesProvider`:
```
public async UniTask<IList<T>> GetAddressablesList<T>(string key, CancellationToken cancellationToken = default)
        {
            //This function does nothing if initialization has already occurred
            await Addressables.InitializeAsync().WithCancellation(cancellationToken);

            AsyncOperationHandle<IList<T>> request = Addressables.LoadAssetsAsync<T>(key, null);
            await request.WithCancellation(cancellationToken);
            return request.Result;
        }
```
   - You can achieve the same with `WaitForBoolean` 
   - Or by introducing `UniTaskCompletionSource`
- It is a really bad idea to mix events (`public event Action<string> OnReceiveMatrixAccessToken;`) with Tasks, this is the root of all the issues in the class. You should have a seamless flow: you can achieve it by introducing  `UniTaskCompletionSource`, it is done correctly in many places by the Social Stream

Reply:
Done, could you check?